pkgs/top-level/php-packages.nix

```diff
@@ -592,41 +583,17 @@ let
     };
   };
 
-  pthreads = if isPhp73 then pthreads32-dev else pthreads32;
-
-  pthreads32 = assert (pkgs.config.php.zts or false); assert !isPhp73; buildPecl rec {
-    version = "3.2.0";
-    pname = "pthreads";
-
-    src = pkgs.fetchFromGitHub {
-      owner = "krakjoe";
-      repo = "pthreads";
-      rev = "v${version}";
-      sha256 = "17hypm75d4w7lvz96jb7s0s87018yzmmap0l125d5fd7abnhzfvv";
-    };
-
-    buildInputs = with pkgs; [ pcre.dev ];
-  };
-
-  pthreads32-dev = assert (pkgs.config.php.zts or false); assert isPhp73; buildPecl {
-    version = "3.2.0-dev";
-    pname = "pthreads";
-
-    src = pkgs.fetchFromGitHub {
-      owner = "krakjoe";
-      repo = "pthreads";
-      rev = "4d1c2483ceb459ea4284db4eb06646d5715e7154";
-      sha256 = "07kdxypy0bgggrfav2h1ccbv67lllbvpa3s3zsaqci0gq4fyi830";
-    };
-
-    buildInputs = with pkgs; [ pcre2.dev ];
-  };
-
```

Comment:
Pthreads and pinba phpPackages removed?

Reply:
For now yes because they did not build. Whenever this change is pulled in and build it's easier to add things back and actually test it in a better way.

We reasoned that we had a set of changes that was big enough as it was.

So for now I think that's best, but I won't reject adding it back in a better way if they build.

Reply:
This pakages builded with this patch

```
diff --git a/pkgs/top-level/php-packages.nix b/pkgs/top-level/php-packages.nix
index 21207c72fe9..7dd422e839d 100644
--- a/pkgs/top-level/php-packages.nix
+++ b/pkgs/top-level/php-packages.nix
@@ -18,6 +18,7 @@ let
     pname = "php-${pname}";
   });

+  isPhp72 = pkgs.lib.versionAtLeast php.version "7.2";
   isPhp73 = pkgs.lib.versionAtLeast php.version "7.3";
   isPhp74 = pkgs.lib.versionAtLeast php.version "7.4";

@@ -295,7 +296,7 @@ in
   # This is a set of PHP extensions meant to be used in php.buildEnv
   # or php.withExtensions to extend the functionality of the PHP
   # interpreter.
-  extensions = {
+  extensions = rec {
     apcu = buildPecl {
       version = "5.1.18";
       pname = "apcu";
@@ -578,6 +579,53 @@ in
       configureFlags = [ "--with-excel" "--with-libxl-incdir=${pkgs.libxl}/include_c" "--with-libxl-libdir=${pkgs.libxl}/lib" ];
     };

+    pinba = if isPhp73 then pinba112 else pinba111;
+
+    pinba111 = buildPecl {
+      version = "1.1.1";
+      pname = "pinba";
+
+      src = pkgs.fetchFromGitHub {
+        owner = "tony2001";
+        repo = "pinba_extension";
+        rev = "RELEASE_1_1_1";
+        sha256 = "1kdp7vav0y315695vhm3xifgsh6h6y6pny70xw3iai461n58khj5";
+      };
+
+      meta = with pkgs.lib; {
+        description = "PHP extension for Pinba";
+        longDescription = ''
+          Pinba is a MySQL storage engine that acts as a realtime monitoring and
+          statistics server for PHP using MySQL as a read-only interface.
+        '';
+        homepage = "http://pinba.org/";
+        broken = (isPhp73);
+      };
+    };
+
+    pinba112 = buildPecl {
+      version = "1.1.2-dev";
+      pname = "pinba";
+
+      src = pkgs.fetchFromGitHub {
+        owner = "tony2001";
+        repo = "pinba_extension";
+        rev = "edbc313f1b4fb8407bf7d5acf63fbb0359c7fb2e";
+        sha256 = "02sljqm6griw8ccqavl23f7w1hp2zflcv24lpf00k6pyrn9cwx80";
+      };
+
+      meta = with pkgs.lib; {
+        description = "PHP extension for Pinba";
+        longDescription = ''
+          Pinba is a MySQL storage engine that acts as a realtime monitoring and
+          statistics server for PHP using MySQL as a read-only interface.
+        '';
+        homepage = "http://pinba.org/";
+
+        broken = (!isPhp73);
+      };
+    };
+
     protobuf = buildPecl {
       version = "3.11.2";
       pname = "protobuf";
@@ -595,6 +643,40 @@ in
       };
     };

+    pthreads = if isPhp73 then pthreads32-dev else pthreads32;
+
+    pthreads32 = buildPecl rec {
+      version = "3.2.0";
+      pname = "pthreads";
+
+      src = pkgs.fetchFromGitHub {
+        owner = "krakjoe";
+        repo = "pthreads";
+        rev = "v${version}";
+        sha256 = "17hypm75d4w7lvz96jb7s0s87018yzmmap0l125d5fd7abnhzfvv";
+      };
+
+      buildInputs = with pkgs; [ pcre.dev ];
+
+      meta.broken = (!isPhp72 || isPhp73);
+    };
+
+    pthreads32-dev = buildPecl {
+      version = "3.2.0-dev";
+      pname = "pthreads";
+
+      src = pkgs.fetchFromGitHub {
+        owner = "krakjoe";
+        repo = "pthreads";
+        rev = "4d1c2483ceb459ea4284db4eb06646d5715e7154";
+        sha256 = "07kdxypy0bgggrfav2h1ccbv67lllbvpa3s3zsaqci0gq4fyi830";
+      };
+
+      buildInputs = with pkgs; [ pcre2.dev ];
+
+      meta.broken = (!isPhp73 || isPhp74);
+    };
+
     redis = buildPecl {
       version = "5.1.1";
       pname = "redis";

```

Reply:
Pinba is now added back in 1e42497447d385a87a3c8df21f269575d2061c3a.

A bit differently written but it's there. I will work on pthreads next.

Reply:
Pthreads builds now for me on php72 and php73 but not php74. And it is archived by upstream: https://github.com/krakjoe/pthreads

So it doesn't seem to be supported or updated anymore. So I'll drop that one for now unless you have a patch for it that works on all versions.

Reply:
Ok, added it as dccccab1e24 and marked it as broken for php74. Now it's back to how it was before :)

Reply:
> Pthreads builds now for me on php72 and php73 but not php74. And it is archived by upstream: https://github.com/krakjoe/pthreads

On php 7.4 package pthreads replaced to https://github.com/krakjoe/parallel

Reply:
I propose to do this in a followup PR. No need to delay this PR any further :-)

Reply:
Yeah, not adding anything new here. But I restored previous state at least :-)